#!/usr/bin/env python

PACKAGE = "static_obstacle_detection"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


gen.add("static_minPoints", int_t, 0, "static_minPoints", 1, 1, 100)
gen.add("static_epsilon", double_t, 0, "static_epsilon", 0.5, 0.0, 20.0)
gen.add("static_minClusterSize", int_t, 0, "static_minClusterSize", 5, 1, 100)
gen.add("static_maxClusterSize", int_t, 0, "static_maxClusterSize", 5000, 1, 10000)

gen.add("static_xMinROI", double_t, 0, "static_xMinROI", 0.5, 0, 10)
gen.add("static_xMaxROI", double_t, 0, "static_xMaxROI", 11.0, -10, 200)
gen.add("static_yMinROI", double_t, 0, "static_yMinROI", -2.5, -200, 10)
gen.add("static_yMaxROI", double_t, 0, "static_yMaxROI", 2.5, -10, 200)
gen.add("static_zMinROI", double_t, 0, "static_zMinROI", -0.78, -10, 0)
gen.add("static_zMaxROI", double_t, 0, "static_zMaxROI", -0.25, -2, 20)

gen.add("static_xMinBoundingBox", double_t, 0, "static_xMinBoundingBox", 0.01, 0, 1000);
gen.add("static_xMaxBoundingBox", double_t, 0, "static_xMaxBoundingBox", 0.5, 0, 3);
gen.add("static_yMinBoundingBox", double_t, 0, "static_yMinBoundingBox", 0.0, 0, 10);
gen.add("static_yMaxBoundingBox", double_t, 0, "static_yMaxBoundingBox", 0.5, 0, 3);
gen.add("static_zMinBoundingBox", double_t, 0, "static_zMinBoundingBox", 0.0, -3, 0.1);
gen.add("static_zMaxBoundingBox", double_t, 0, "static_zMaxBoundingBox", 0.5, 0, 3);

exit(gen.generate(PACKAGE, "static_obstacle", "static_hyper_parameter"))
